data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/leader/InstanceApiStrategy.java

```diff
@@ -37,9 +37,13 @@ public DbMetadata describeDb(String dbIdentifier) {
                 .dbInstanceIdentifier(dbIdentifier)
                 .build();
 
-        final DescribeDbInstancesResponse response = rdsClient.describeDBInstances(request);
-        final DBInstance dbInstance = response.dbInstances().get(0);
-        return new DbMetadata(dbIdentifier, dbInstance.endpoint().address(), dbInstance.endpoint().port());
+        try {
+            final DescribeDbInstancesResponse response = rdsClient.describeDBInstances(request);
+            final DBInstance dbInstance = response.dbInstances().get(0);
```

Comment:
What is the behavior for 2/3 AZ deployment ?

Reply:
2 AZ has 1 active instance, so this will return the info for that instance.
3 AZ is a cluster, the ClusterApiStrategy will handle and return primary instance info.